Sources/MapboxMaps/Foundation/Extensions/UIWindow.swift

```diff
@@ -0,0 +1,16 @@
+import UIKit
+
+@available(iOS 13.0, *)
+extension UIWindow {
+    private static let templateApplicationSceneSelector = Selector(("templateApplicationScene"))
+
+    internal var parentScene: UIScene? {
+        var carPlayScene: UIScene? {
+            guard self.responds(to: UIWindow.templateApplicationSceneSelector) else {
+                return nil
+            }
+            return self.value(forKey: UIWindow.templateApplicationSceneSelector.description) as? UIScene
+        }
+        return windowScene ?? carPlayScene
+    }
```

Comment:
This was a tricky one - we don't really know at the compile time if CarPlay is available. So it seems that resorting to runtime checks is the way to go. 

UIKit provides us only with `windowScene`, which is `nil` in case of CarPlay window. `CPWindow` has a separate scene property called `templateApplicationScene`. 
With the help of Objective C runtime we can access here the car play scene(if any).
Thus the return value of this computed property will be `windowScene` in case of `UIWindow`, and `templateApplicationScene` in case of `CPWindow`. 

I'm open to suggestions on how to improve this :)

Reply:
```
    internal var parentScene: UIScene? {
        return windowScene ?? (self as? CPWindow)?.templateApplicationScene
    }
```

Does this work?

Reply:
It's fine to import CarPlay - the framework will be weak-linked automatically if needed based on the deployment target, and we require Xcode 12.5 or later to build, so CarPlay will always be available when linking.

Reply:
Good tip, looks much better now!